pkg/apis/camel/v1/trait/cron.go

```diff
@@ -27,8 +27,11 @@ package trait
 // Integrations that start from the following components are evaluated by the cron trait: `timer`, `cron`, `quartz`.
 //
 // The rules for using a Kubernetes CronJob are the following:
-//   - `timer`: when periods can be written as cron expressions. E.g. `timer:tick?period=60000`.
+//
+//   - `timer`: when period is set in milliseconds with no remaining seconds, for example 120000. If there is any second left as in 121000 (120s and 1s) or the presence of any of these parameters (delay, repeatCount, time) then a CronJob  won't be created, but a standard deployment.
```

Comment:
Just out of curiosity. Where is this behavior defined? in the runtime?

Reply:
It is defined partly in the cron.go trait and the camel-k-cron (camel-k-runtime).